autorest/codegen/serializers/general_serializer.py

```diff
@@ -175,7 +193,11 @@ def serialize_version_file(self) -> str:
 
     def serialize_setup_file(self) -> str:
         template = self.env.get_template("setup.py.jinja2")
-        params = {}
+        params: Dict[str, Any] = {"is_legacy": self.code_model.is_legacy}
```

Comment:
I think for all serializers we don't need to pass `is_legacy` separately. We already pass the code model to pretty much every template, so you can just call `code_model.is_legacy` already in the template

Reply:
I updated to use code_model in template, and added code_model parameter to render here: https://github.com/Azure/autorest.python/pull/1236/files#diff-780f04d7de5ff34a70c79a2ed24279e346ad2aafcaae6a07c9aaac89d2be49f1R181